text/0000-history.md

```diff
@@ -0,0 +1,83 @@
+# Meta
+[meta]: #meta
+- Name: Buildpack layer metadata
+- Start Date: 2021-12-02
+- Author(s): [@samj1912](https://github.com/samj1912)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Add layer history metadata to the output image for buildpack and app layers to improve visualization in tools like `DockerHub`, `dive` etc.
+
+# Definitions
+[definitions]: #definitions
+
+`history`: The history key in the `OCI` config. See [properties](https://github.com/opencontainers/image-spec/blob/main/config.md#properties)
+
+# Motivation
+[motivation]: #motivation
+
+This will allow Buildpack built images to be better visualized by common container visualization and debuggin tools.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The following keys will be added to the following files - 
+
+## launch.toml
+
+```
+[[slices]]
+# name is a new key which uniquely identitifies the app slice
+# required key
+name = "<name-of-the-slice>" # max 255 chars
+```
+
+All these tools look at `history.created_by` for each layer to visualize the layer. We should populate this key with a value that describes where a layer came from, for buildpacks this can be -
+
+- `{{ buildpack.id }}: Buildpack: {{ buildpack.name }} Layer: {{ buildpack.layer.name }}`
```

Comment:
`Layer: {{ buildpack.layer.name }}, Created by: {{ buildpack.id }}`

Reply:
The lifecycle stringifies buildpacks as "Name Version" and sometimes "ID@Version" ...those could be other options.